### ISSUE TITLE
Testing: Enable all accessibility rules validation in e2e tests

### ISSUE THREAD

**Issue by gziolo** [WordPress/gutenberg#15452]
All e2e tests have accessibility rules validation enabled by default as of #15018.

> This PR enables [Axe](https://www.deque.com/axe/) tests for every individual e2e test containing the block editor. They will run at the end of each test and perform static accessibility related analysis based on the DOM present in the test. It isn't perfect, but it should help to catch basic regressions moving forward.

We still need to enable the following rules:
- [ ] `aria-allowed-role`
- [ ] `aria-hidden-focus`
- [ ] `aria-input-field-name`
- [ ] `aria-valid-attr-value`
- [ ] `button-name`
- [ ] `color-contrast`
- [ ] `dlitem`
- [ ] `duplicate-id`
- [ ] `label`
- [ ] `link-name`
- [ ] `listitem`
- [ ] `region`

All rules are listed and explained in depth here:
https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md

There are ongoing efforts to fix issues reported as part Accessibility Audit which might resolve the majority of violations. See the related project: https://github.com/WordPress/gutenberg/projects/25.

To enable one of the listed rules, the following file should be updated:
https://github.com/WordPress/gutenberg/blob/a797ec5567d786604081785cf0e15620dee903dc/packages/e2e-tests/config/setup-test-framework.js#L163-L174

In many cases, it might require additional work to ensure that violations are resolved. That's why I recommend opening 1 PR per rule.

**Comment by gziolo** [comment]
It's no longer relevant after https://github.com/WordPress/gutenberg/pull/26626. We are switching to running axe verification on demand.
